manifests/proxy.pp

```diff
@@ -549,34 +550,54 @@
   }
 
   # Controlling the 'zabbix-proxy' service
-  service { $proxy_service_name:
-    ensure     => running,
-    enable     => $enable,
-    hasstatus  => true,
-    hasrestart => true,
-    require    => [
-      Package["zabbix-proxy-${db}"],
-      File[$include_dir],
-      File[$proxy_configfile_path]],
+  if $manage_service {
+    service { $proxy_service_name:
+      ensure     => running,
+      hasstatus  => true,
+      hasrestart => true,
+      subscribe  => [
+        File[$proxy_configfile_path],
+        Class['::zabbix::database']
+        ],
+      require    => [
+        Package["zabbix-proxy-${db}"],
+        File[$include_dir],
+        File[$proxy_configfile_path],
+        Class['::zabbix::database']
+        ],
+    }
   }
 
   # if we want to manage the databases, we do
   # some stuff. (for maintaining database only.)
-  if $manage_database == true {
-    class { '::zabbix::database':
-      database_type     => $database_type,
-      zabbix_type       => 'proxy',
-      database_name     => $database_name,
-      database_user     => $database_user,
-      database_password => $database_password,
-      database_host     => $database_host,
-      zabbix_proxy      => $zabbix_proxy,
-      zabbix_proxy_ip   => $zabbix_proxy_ip,
-      notify            => Service[$proxy_service_name],
-      before            => [
-        Service[$proxy_service_name],
-        Class["zabbix::database::${database_type}"],
-        ],
+  if $manage_database  {
+    if $manage_service {
+      class { '::zabbix::database':
+        database_type     => $database_type,
+        zabbix_type       => 'proxy',
+        database_name     => $database_name,
+        database_user     => $database_user,
+        database_password => $database_password,
+        database_host     => $database_host,
+        zabbix_proxy      => $zabbix_proxy,
+        zabbix_proxy_ip   => $zabbix_proxy_ip,
+        before            => [
+          Service[$proxy_service_name],
```

Comment:
is this the only different line? We shouldt do code duplication here. Please write Service[$proxy_service_name] into a variable and use that in the before parameter.